src/analyses/basePriv.ml

```diff
@@ -250,7 +250,7 @@ struct
 
   let invariant_global ask getg = function
     | `Right g' -> (* global *)
-      ValueDomain.invariant_global (read_unprotected_global getg) g'
+      ValueDomain.invariant_global (read_unprotected_global getg) g' (* TODO: disjunct with mutex_inits instead of join? *)
```

Comment:
Using duplicate versions of `read_unprotected_global` and `get_mutex_global_x_with_mutex_inits` which operate on invariant expressions rather than `ValueDomain`, we could make these invariants maybe more precise.

I just don't know if it's worth it though. Currently the logic is shared with the usual `read_global` used by the analysis. Duplicating would require them to be maintained in sync.

Reply:
I will change the TODO into a comment indicating a bit more precisely what could be made more precise here (and a few other similar places).